mesonbuild/interpreter/interpreter.py

```diff
@@ -674,19 +674,12 @@ def extract_variables(self, kwargs: T.Dict[str, T.Union[T.Dict[str, str], T.List
     )
     def func_declare_dependency(self, node: mparser.BaseNode, args: T.List[TYPE_var],
                                 kwargs: kwargs.FuncDeclareDependency) -> dependencies.Dependency:
-        deps = kwargs['dependencies']
         incs = self.extract_incdirs(kwargs)
-        libs = kwargs['link_with']
-        libs_whole = kwargs['link_whole']
-        sources = self.source_strings_to_files(kwargs['sources'])
-        compile_args = kwargs['compile_args']
-        link_args = kwargs['link_args']
         variables = kwargs['variables']
         version = kwargs['version']
+        sources = self.source_strings_to_files(kwargs['sources'])
         if version is None:
             version = self.project_version
-        d_module_versions = kwargs['d_module_versions']
-        d_import_dirs = self.extract_incdirs(kwargs, 'd_import_dirs')
```

Comment:
Intermediate variables being useless is kind of a matter of opinion, it can in fact be argued it makes the InternalDependency() easier to read.

However, one thing is definitely outright broken. You stop using extract_incdirs on d_import_dirs, then you add it back in in the next commit when "mypy spots that it is an issue".

Reply:
To add to the confusion, you also misspelled it in the double refactor. `d_import_dirs` -> `d_module_dirs`